trie-db/src/proof/generate.rs

```diff
@@ -75,13 +88,24 @@ impl<'a, C: NodeCodec> StackEntry<'a, C> {
 
 	/// Encode this entry to an encoded trie node with data properly omitted.
 	fn encode_node(mut self) -> TrieResult<Vec<u8>, C::HashOut, C::Error> {
+		let omit_value = self.omit_value;
 		let node_data = self.node.data();
-		Ok(match self.node.node_plan() {
+		let value_with_omission = |
+			value_range: ValuePlan,
+		| -> Option<Value>
+		{
+			if matches!(omit_value, ValueMgmt::Standard) {
+				Some(value_range.build(&node_data))
+			} else {
+				None
+			}
+		};
+		let encoded = match self.node.node_plan() {
 			NodePlan::Empty => node_data.to_vec(),
-			NodePlan::Leaf { .. } if !self.omit_value => node_data.to_vec(),
+			NodePlan::Leaf { .. } if matches!(omit_value, ValueMgmt::Standard) => node_data.to_vec(),
```

Comment:
I'm not sure I see the point for the 3 variants enum, because in this code we only match against Standard, both here and just above line 97.
why not just keeping a bool here ?